testing/test_recwarn.py

```diff
@@ -8,25 +8,19 @@
 def test_recwarn_functional(testdir):
     reprec = testdir.inline_runsource("""
         import warnings
-        oldwarn = warnings.showwarning
```

Comment:
This is test and `test_recording` below were testing an implementation detail which is no longer true in 3.6, because `catch_warnings` has [changed significantly since `3.5`](https://github.com/python/cpython/blob/master/Lib/warnings.py#L457).